test/spec/modules/yieldbotBidAdapter_spec.js

```diff
@@ -161,112 +254,225 @@ describe('Yieldbot adapter tests', function() {
-
       expect(window.yieldbot._initialized).to.equal(true);
       sinon.assert.calledWith(yieldbotLibStub.defineSlot, 'medrec');
       sinon.assert.calledWith(yieldbotLibStub.defineSlot, 'leaderboard');
 
-      window.pbjs._bidsRequested = window.pbjs._bidsRequested.filter(o => {
-        return o.bidderCode !== 'yieldbot';
-      });
+      removeBids('yieldbot');
 
-      const refreshBids = bidderRequest.bids.filter((object) => { return object.placementCode === '/4294967296/adunit1'; });
-      let refreshRequest = Object.assign({}, bidderRequest);
+      const refreshBids = localRequest.bids.filter((object) => { return object.placementCode === '/4294967296/adunit1'; });
+      let refreshRequest = cloneJson(localRequest);
       refreshRequest.bids = refreshBids;
       expect(refreshRequest.bids.length).to.equal(1);
 
       adapter.callBids(refreshRequest);
       mockYieldbotBidRequest();
 
-      const bid = refreshBids[0];
-      const expectedSlots = { 'leaderboard': [[728, 90], [970, 90]] };
+      const expectedSlots = { 'leaderboard': [['728', '90'], ['970', '90']] };
 
       sinon.assert.calledWithExactly(yieldbotLibStub.nextPageview, expectedSlots);
     });
 
-    it('should not throw on callBids without bidsRequested', function() {
-      const adapter = new YieldbotAdapter();
-      adapter.callBids(bidderRequest);
+    it('should not repeat multiply defined slot sizes', function() {
+      adapter.callBids(localRequest);
       mockYieldbotBidRequest();
 
+      sinon.assert.calledOnce(yieldbotLibStub.nextPageview);
+      const expectedSlots = { 'leaderboard': [['728', '90'], ['970', '90']], 'medrec': [['300', '250'], ['300', '600']]};
+      sinon.assert.calledWithExactly(yieldbotLibStub.nextPageview, expectedSlots);
+    });
+
+    /**
+     * Keep test for regression. Yieldbot adapter no longer refers to $$PREBID_GLOBAL$$._bidsRequested
+     * @private
+     */
+    it('should not throw on callBids without bidsRequested', function() {
       expect(window.yieldbot._initialized).to.equal(true);
 
-      window.$$PREBID_GLOBAL$$._bidsRequested = window.$$PREBID_GLOBAL$$._bidsRequested.filter(o => {
-        return o.bidderCode !== 'yieldbot';
-      });
+      removeBids('yieldbot');
 
-      adapter.callBids(bidderRequest);
+      adapter.callBids(cloneJson(localRequest));
       mockYieldbotBidRequest();
       sinon.assert.calledOnce(yieldbotLibStub.nextPageview);
     });
 
     it('should not add empty bidResponse on callBids without bidsRequested', function() {
-      window.$$PREBID_GLOBAL$$._bidsRequested = window.$$PREBID_GLOBAL$$._bidsRequested.filter(o => {
-        return o.bidderCode !== 'yieldbot';
-      });
+      expect(window.yieldbot._initialized).to.equal(true);
 
-      const adapter = new YieldbotAdapter();
-      adapter.callBids(bidderRequest);
-      mockYieldbotBidRequest();
+      removeBids('yieldbot');
 
       let bidResponses = window.$$PREBID_GLOBAL$$._bidsReceived.filter(o => {
```

Comment:
you should not access pbjs private vars. you can test this by stubbing `bidmanager.addBidResponse`